### PR TITLE
create a modal and new trip form

### LINKED ISSUE #58: Generic Modal

### LINKED ISSUE #60: New Trip Form

### PR DESCRIPTION
**Pull request by NLmeng**
closes #60 closes #58

### CODE REVIEW

File: client/src/components/MapElement/MapElement.js [hunk 1]
```diff
@@ -11,32 +13,88 @@ MapElement.propTypes = {
...
+  const dispatch = useDispatch()
+
+  const [destination, setDestination] = useState('')
+  const [stagesPerDay, setStagesPerDay] = useState('')
+  const [budget, setBudget] = useState('')
+  const [numberOfDays, setNumberOfDays] = useState('')
+  const [invalidForm, setInvalidForm] = useState(false)
+
+  const resetForm = () => {
+    setDestination('')
```
Comment (Graywing13) [inline client/src/components/MapElement/MapElement.js]: Do we want to consider using react-hook-form?

File: client/src/components/MapElement/MapElement.js [hunk 2]
```diff
@@ -11,32 +13,88 @@ MapElement.propTypes = {
...
+        title='Manifesting A New Trip...'
+        footer={
           <Button
-            onClick={() => {}}
-            className='font-bolder mt-4 bg-slate-300 hover:bg-slate-400'
+            className='font-bolder mt-4 w-full bg-slate-300 hover:bg-slate-400'
+            onClick={handleSubmitNewTripForm}
+            type='submit'
           >
             Brew 🍵
```
Comment (Graywing13) [inline client/src/components/MapElement/MapElement.js]: Is it possible to make this button narrower, to match the inputs? 

image

Comment (Graywing13) [reply client/src/components/MapElement/MapElement.js]: And if it's not too hard, make the rounding of the modal & button match (if it's not possible bc of MUI then it's ok)

File: client/src/components/MapElement/MapElement.js [hunk 3]
```diff
@@ -11,32 +13,88 @@ MapElement.propTypes = {
...
+
+  const resetForm = () => {
+    setDestination('')
+    setStagesPerDay('')
+    setBudget('')
+    setNumberOfDays('')
+    setInvalidForm(false)
+  }
+  const handleCloseNewTripModal = () => {
+    dispatch(setAppView(AppView.TRIP_VIEW))
```
Comment (Graywing13) [inline client/src/components/MapElement/MapElement.js]: Ah I get what you mean now. 

Maybe if they exit by clicking the bg, we should stay in the globe mode but just close the modal? And only go to TRIP_VIEW if they press "brew/submit"?

File: client/src/components/MapElement/MapElement.js (hunk 1 above)
Comment (NLmeng) [reply client/src/components/MapElement/MapElement.js]: probably should, looks nicer

File: client/src/components/MapElement/MapElement.js (hunk 3 above)
Comment (NLmeng) [reply client/src/components/MapElement/MapElement.js]: i added this, not sure if i broke sth tho

File: client/src/components/MapElement/MapElement.js
Comment (Graywing13) [inline client/src/components/MapElement/MapElement.js:1]: Sorry I realize this is my bad for putting the old code here... should we provide better encapsulation if MapElement was used for the map, and move the new trip logic somewhere else?

File: client/src/components/common/Modal.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        onClose={handleClose}
+      >
+        <Box m='0'>
+          <DialogTitle
+            id='draggable-dialog-title'
+            fontSize='2rem'
+            className='cursor-grab bg-slate-100'
+          />
+          <Box my='0.5rem' p='0'>
+            <h1
```
Comment (Graywing13) [inline client/src/components/common/Modal.js:73]: Can we try a little more space above the title, e.g. pt-4 or pt-2?

**Review by Graywing13 — COMMENTED** [review]
Setup looks clean 👏 well-documented too 🏕️ just some stylistic stuff to consider

File: client/src/components/common/Modal.js (hunk 4 above)
Comment (Graywing13) [reply client/src/components/common/Modal.js:73]: just so there's a bit more of a distinction between the draggable section and the title. 

maybe make the draggable slate-200 if it helps? if not then the current is ok too

File: client/src/components/MapElement/MapElement.js
Comment (NLmeng) [reply client/src/components/MapElement/MapElement.js:1]: fixed

File: client/src/components/MapElement/MapElement.js [hunk 5]
```diff
@@ -96,7 +23,7 @@ export function MapElement({ className }) {
         </h1>
       </span>
     )
-  }, [appView, activeTripId, newTripForm])
+  }, [appView, activeTripId, NewTripForm])
```
Comment (Graywing13) [inline client/src/components/MapElement/MapElement.js]: huh its curious that we have to watch an import. but if thats needed then it's okay

**Review by Graywing13 — APPROVED** [review]
lgtm 🏕️

**Pull request merged**